File: Algorithm/数组/长度最小子数组/minSubArrayLen.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution
{
public:
    int minSubArrayLen(vector<int> &nums, int s){
        int minLen = INT_MAX;
        int sum = 0;
        int left = 0;
        int subLength = 0;
        for (int left = 0; left < nums.size(); left++)
        {
            sum += nums[left];
            while (sum >= s)
            {
                subLength = left - left + 1;
                minLen = minLen < subLength ? minLen : subLength;
                sum -= nums[left++];
            }
        }
        return minLen == INT_MAX ? 0 : minLen;
    }
};

int main()
{
    vector<int> nums={2,3,1,2,4,3};
    int s=7;
    Solution solution;
    cout << solution.minSubArrayLen(nums, s) << endl;
    return 0;
}

File: Algorithm/leetcode/practice/数组/长度最小子数组/minSubArrayLen.py
class Solution:
    def minSubArrayLen(self, nums: list[int], s: int) -> int:
        left = 0
        right = 0
        sum = 0
        minLen = len(nums) + 1
        while right < len(nums):
            sum += nums[right]
            while sum >= s:
                minLen = min(minLen, right - left + 1)
                sum -= nums[left]
                left += 1
            right += 1
        return 0 if minLen == len(nums) + 1 else minLen

if __name__=='__main__':
    print(Solution().minSubArrayLen([2,3,1,2,4,3], 7))